### ISSUE TITLE
atlas-api chat template uses wrong special tokens for OLMo-3-Think

### ISSUE THREAD

**Issue by web3guru888** [web3guru888/ATLAS#15]
## Summary

`ChatCompletionRequest::to_prompt()` in `atlas-api` hardcodes `<|system|>`, `<|user|>`, and `<|assistant|>` as chat delimiters. OLMo-3-7B-Think uses `<|im_start|>`/`<|im_end|>` (ChatML format), which are completely different tokens. This causes the model to see unknown/wrong tokens, producing incoherent output through the API.

## Current Code

```rust
// crates/atlas-api/src/types.rs:105
pub fn to_prompt(&self) -> String {
    let mut prompt = String::new();
    for msg in &self.messages {
        match msg.role.as_str() {
            "system"    => { prompt.push_str(&format!("<|system|>\n{}\n", msg.content)); }
            "user"      => { prompt.push_str(&format!("<|user|>\n{}\n<|assistant|>\n", msg.content)); }
            "assistant" => { prompt.push_str(&format!("{}\n", msg.content)); }
            _           => { prompt.push_str(&format!("{}\n", msg.content)); }
        }
    }
    prompt
}
```

## Expected (OLMo-3-Think ChatML)

```
<|im_start|>system\nYou are a helpful assistant.<|im_end|>\n<|im_start|>user\nWhat is the capital of France?<|im_end|>\n<|im_start|>assistant\n
```

OLMo-3 special tokens:
- `<|im_start|>` = 100264
- `<|im_end|>` = 100265
- `<|endoftext|>` = 100257 (EOS)

## Impact

- `<|system|>`, `<|user|>`, `<|assistant|>` are NOT in OLMo-3's vocabulary as special tokens
- The tokenizer encodes them as regular BPE subwords, producing wrong token sequences
- The model has never seen these tokens during training → incoherent generation
- This is why the `atlas api serve` endpoint produces garbage with OLMo-3-7B-Think even with the v4.0.4 tokenizer fix

### Option A: Model-specific chat templates (recommended)
Read `tokenizer_config.json` which contains the `chat_template` (Jinja2 format). Parse it or detect the model family:

```rust
enum ChatTemplate {
    ChatML,     // OLMo-3, Qwen: <|im_start|>/<|im_end|>
    Llama3,     // Llama-3: <|begin_of_text|><|start_header_id|>...
    Default,    // Generic: <|system|>/<|user|>/<|assistant|>
}
```

### Option B: Read from tokenizer_config.json
```json
{"chat_template": "{% for message in messages %}..."}
```
Parse the Jinja2 template to extract the special token pattern.

### Option C: Configurable via API config
Let the user specify the chat template format in the server config.

## Verification

Confirmed in `gpu_olmo3_tokenizer_e2e` test (v4.0.4):
- Raw text prompt → greedy produces "a a a a..." (flat logits, 0.96% max prob)
- Chat template `<|im_start|>user\n...` → diverse tokens (30/30 unique)
- The model IS working — it just needs the right framing

## Related
- Issue #13 (EOS dead code) — the model can't signal completion
- Issue #12 (tokenizer fix) — resolved in v4.0.4

## Environment
- ATLAS v4.0.4
- Affects atlas-api serve with OLMo-3-7B-Think
- SmolLM2 uses yet another template format

**Comment by web3guru888** [comment]
Fixed in commit e831935.

**Changes:**
- Added `ChatTemplate` enum: ChatML, Llama3, Generic (default: ChatML)
- ChatML: `<|im_start|>role\ncontent<|im_end|>\n` (OLMo-3, SmolLM2, Qwen)
- Auto-detects template from tokenizer special tokens
- Added `chat_template` to `InferState`, handler uses it for prompt formatting
- `to_prompt()` defaults to ChatML; `to_prompt_with()` takes explicit template

**Tests:** `to_prompt_chatml`, `to_prompt_llama3`, `to_prompt_generic`, `detect_template_*` ✅
